test/dynamo/test_modes.py

```diff
@@ -324,6 +324,40 @@ def fn(x):
             fn(inp)
         self.assertEqual(cnt.frame_count, 2)
 
+    def test_torch_function_mode_highest_priority(self):
+        class TestSubclass(torch.Tensor):
+            @classmethod
+            def __torch_function__(cls, func, types, args, kwargs=None):
+                if not kwargs:
+                    kwargs = {}
+                if func == torch.add:
+                    return torch.ones(2, 2)
+                return super().__torch_function__(func, types, args, kwargs)
+
+        class TestMode(BaseTorchFunctionMode):
```

Comment:
Can you test nested TorchFunctionMode? We should either handle it or it should graph break.

Reply:
Yes I can add a test for this, we handle this.